src/fastmath/ml/regression.clj

```diff
@@ -1246,3 +1269,52 @@
      {:dose ##Inf
       :p p
       :stderr ##Inf})))
+
+(defn- residuals-analysis
+  "Summary statistics used for printing residuals with model"
+  [residuals]
+  [{:min (apply min residuals)
+    :1q (stats/quantile residuals 0.25)
+    :median (stats/median residuals)
+    :3q (stats/quantile residuals 0.75)
+    :max (apply max residuals)}])
+
+(defn- residuals-label
+  [model weights]
+  (let [weighted? (->> (or (:weights weights) weights)
+                       (some (fn [x] (not (= 1.0 x)))))
```

Comment:
@genmeblog here's my main change! Looking at the weights list/vector and using that to add to the label